xdsl/ir/core.py

```diff
@@ -807,7 +814,7 @@ def __repr__(self) -> str:
             f"<{self.__class__.__name__}[{self.type}]"
             f" index: {self.index},"
             f" operation: {self.op.name},"
-            f" uses: {self.uses.get_length()}>"
+            f" uses: {tuple(self.uses)}>"
```

Comment:
why tuple and not a repr on the uses object directly?

Reply:
Also the current implementation of get_length is ridiculous, it would be great if you could open a separate PR computing it in a less expensive way, if you happen to have the time to try that, and seeing if the test you added in this PR passes without it.

Reply:
OK but then this change seems separate and deserves its own PR?